src/analyzing/Implicit_return.ml

```diff
@@ -36,39 +36,41 @@ open AST_generic
 let rec mark_first_instr_ancestor (cfg : IL.cfg) i =
   let node = cfg.graph#nodes#find i in
   match node.n with
+  (* If control reaches a catch node, it's an exception. It can't be a return,
+   * so stop visiting here.
+   *)
+  | NOther (Noop "catch") -> ()
+  (* Visit ancestor for exit, noop, goto, and join nodes. *)
   | Exit
   | NOther (Noop _)
   | NGoto _
   | Join ->
       CFG.predecessors cfg i
       |> List.iter (fun (pred_i, _) -> mark_first_instr_ancestor cfg pred_i)
+  (* Certain instruction nodes may be implicitly returned. *)
   | NInstr instr -> (
       match instr with
-      | { i = Assign (_, { eorig = SameAs e; _ }); _ } ->
+      | { i = Assign (_, { eorig = SameAs e; _ }); _ }
+      | { i = Call _; iorig = SameAs e } ->
           e.is_implicit_return <- true
       | _else_ -> ())
   | _else_ -> ()
 
-let mark_implicit_return_nodes_one_function (cfg : IL.cfg) =
-  (* Traverse backward from exit and mark the expression in the
-   * first instruction node along each path.
-   *)
-  mark_first_instr_ancestor cfg cfg.exit
-
 (*****************************************************************************)
 (* Entry point *)
 (*****************************************************************************)
 
 let lang_supports_implicit_return (lang : Lang.t) =
   match lang with
+  | Elixir
```

Comment:
Should we also support Scala?

Reply:
Sure. Added scala tests. Lots of them are passing, but some aren't, e.g. the IL has FIXME nodes, so for those test cases, I made them TODOs, and I'll create a ticket to track making all test cases work for scala.